omnigibson/reward_functions/grasp_reward.py

```diff
@@ -0,0 +1,72 @@
+import math
+from omnigibson.reward_functions.reward_function_base import BaseRewardFunction
+from omnigibson.utils.motion_planning_utils import detect_robot_collision_in_sim
+import omnigibson.utils.transform_utils as T
+
+
+class GraspReward(BaseRewardFunction):
+    """
+    Grasp reward
+    """
+
+    def __init__(self, obj_name, dist_coeff, grasp_reward):
+        # Store internal vars
+        self.prev_grasping = False
+        self.obj_name = obj_name
+        self.obj = None
+        self.dist_coeff = dist_coeff
+        self.grasp_reward = grasp_reward
+
+        # Run super
+        super().__init__()
+
+    def _step(self, task, env, action):
+        self.obj = env.scene.object_registry("name", self.obj_name) if self.obj is None else self.obj
+
+        robot = env.robots[0]
+        obj_in_hand = robot._ag_obj_in_hand[robot.default_arm]
+        current_grasping = True if obj_in_hand == self.obj else False
+        
+        # Reward varying based on combination of whether the robot was previously grasping the desired and object
+        # and is currently grasping the desired object
+
+        # not grasping -> not grasping = Distance between eef and obj reward
+        # not grasping -> grasping = Minimizing MOI + grasp reward
+        # grapsing -> not grapsing = Distance between eef and obj reward
+        # grasping -> grasping = Minimizing MOI + grasp reward
+
+        reward = None
+
+        if not self.prev_grasping and not current_grasping:
```

Comment:
can we make this more concise? there should only be two logic streams right, not four?